### ISSUE TITLE
could not open read-only [file=/home/tey/.questdb/db/stock_history/default/change_percent.d

### ISSUE THREAD

**Issue by b0123456789** [questdb/questdb#2189]
### To reproduce

1.Open web console
2.Run sql  `SELECT * from 'stock_history'`

### Environment

```markdown
- **QuestDB 6.4**:
- **OS Fedora 36**:
```

### Additional context

DDL:

CREATE TABLE 'stock_history' (
  ts TIMESTAMP,
  code SYMBOL capacity 256 CACHE index capacity 256,
  symbol STRING,
  market_type INT,
  name STRING,
  change_percent DOUBLE,
  settlement DOUBLE,
  turnover_ratio DOUBLE,
  volume LONG,
  amount LONG,
  market_total_capitalization DOUBLE,
  created_day SYMBOL capacity 256 CACHE index capacity 256,
  created_at  STRING
) timestamp (ts);

Questdb data from home.

[questdb.zip](https://github.com/questdb/questdb/files/8839483/questdb.zip)

**Comment by puzpuzpuz** [comment]
Could you include a full error log message? It should contain an OS error code in square brackets. Most likely the QuestDB process lacks file permissions or something as trivial.

**Comment by b0123456789** [comment]
where is the log file?

**Comment by b0123456789** [comment]
questdb.zip include all filles.

**Comment by b0123456789** [comment]
[stdout-2022-06-05T18-02-46.zip](https://github.com/questdb/questdb/files/8839757/stdout-2022-06-05T18-02-46.zip)

File from ~/User/questdb./log

**Comment by puzpuzpuz** [comment]
The error code is 2:
```
io.questdb.cairo.CairoException: [2] could not open read-only [file=/home/tey/.questdb/db/stock_history/default/change_percent.d]
```
which means that the file doesn't exist. I don't see any other meaningful errors in the log. Could it be the case that you were doing `kill -9` for the QuestDB process or maybe observing a crash? That could be the reason for the column file being absent.

**Comment by b0123456789** [comment]
`kill -9` is  not the reason, I had try, it work well.
it happend  sometime when doing a lots of inserting .

**Comment by puzpuzpuz** [comment]
> it happend sometime when doing a lots of inserting .

Are you be able to reproduce it? As for this particular table, you should try recreating it if that's possible.

**Comment by b0123456789** [comment]
Yes I can, recreating is ok after removing `~/$user/.questdb` and recreating table.
But it still  happend sometime when doing a lots of inserting .

**Comment by puzpuzpuz** [comment]
Could share a minimal reproducer for this issue?

**Comment by puzpuzpuz** [comment]
Also, can you see any errors in the server logs while you're doing inserts?

**Comment by b0123456789** [comment]
server logs ?  where is it ?

**Comment by b0123456789** [comment]
Content is same in   `~/$user/questdb./log/xx.log`

**Comment by puzpuzpuz** [comment]
In the previously provided log file, I can see that you're running UPDATE statements in parallel with INSERTs. Also, TRUNCATE TABLE. Can you reproduce this issue reliably? If so, what do you do to make it happen?

**Comment by b0123456789** [comment]
Only insert a lot of data.

**Comment by dhiravira** [comment]
I've recently run into this issue on a mac and I suspect it's because of lots of open files. I'm still trying to figure out how to configure that properly but basically I had ingested ~22 million rows and it was partitioned by day and around 20k files were open by questdb at one time (I checked with `lsof`) when doing a full table scan.

**Comment by puzpuzpuz** [comment]
@dhiravira you should try increasing OS X limits for max open files and maybe mmap areas.
